tests/catch/include/hip_test_common.hh

```diff
@@ -171,7 +172,59 @@ static inline void HIP_SKIP_TEST(char const* const reason) noexcept {
   // ctest is setup to parse for "HIP_SKIP_THIS_TEST", at which point it will skip the test.
   std::cout << "Skipping test. Reason: " << reason << '\n' << "HIP_SKIP_THIS_TEST" << std::endl;
 }
+
+/**
+ * @brief Helper template that returns the expected arguments of a kernel.
+ *
+ * @return constexpr std::tuple<FArgs...> the expected arguments of the kernel.
+ */
+template <typename... FArgs> std::tuple<FArgs...> getExpectedArgs(void(FArgs...)){};
+
+/**
+ * @brief Asserts that the types of the arguments of a function match exactly with the types in the
+ * function signature.
+ * This is necessary because HIP RTC does not do implicit casting of the kernel
+ * parameters.
+ * In order to get the kernel function signature, this function should only called when
+ * RTC is disabled.
+ *
+ * @tparam F the kernel function
+ * @tparam Args the parameters that will be passed to the kernel.
+ */
+template <typename F, typename... Args> void validateArguments(F f, Args...) {
+  using expectedArgsTuple = decltype(getExpectedArgs(f));
+  static_assert(std::is_same<expectedArgsTuple, std::tuple<Args...>>::value,
+                "Kernel arguments types must match exactly!");
+}
+
+/**
+ * @brief Launch a kernel using either HIP or HIP RTC.
+ *
+ * @tparam Typenames A list of typenames used by the kernel (unused if the kernel is not a
+ * template).
+ * @tparam K The kernel type. Expects a function or template when RTC is disabled. Expects a
+ * function pointer instead when RTC is enabled.
+ * @tparam Dim Can be either dim3 or int.
+ * @tparam Args A list of kernel arguments to be forwarded.
+ * @param kernel The kernel to be launched (defined in kernels.hh)
+ * @param numBlocks
+ * @param numThreads
+ * @param memPerBlock
+ * @param stream
+ * @param packedArgs A list of kernel arguments to be forwarded.
+ */
+template <typename... Typenames, typename K, typename Dim, typename... Args>
+void launchKernel(K kernel, Dim numBlocks, Dim numThreads, std::uint32_t memPerBlock,
+                  hipStream_t stream, Args&&... packedArgs) {
+#ifndef RTC_TESTING
+    validateArguments(kernel, packedArgs...);
```

Comment:
Looks like these two lines should be tab width = 2 to match the rest of the code